src/test/java/nitezh/ministock/dataaccess/CoinMarketCapRepoTest.java

```diff
@@ -19,11 +20,24 @@
     CoinMarketCapRepo coinMarketCapRepo;
 
     @Test
-    public void getAPIData() throws JSONException{
-        String id = "bitcoin";
+    public void getAllAPIData() throws JSONException{
+        String blankID = "";  //should get all the data if no id is provided
```

Comment:
this only tests for top 100 cryptocurrencies. test whether you can retrieve all 1500 values

Add another test to validate each individual variable within the JSON Object/Array. You may retrieve bitcoin but the prices,marketcap or volume may be empty.